src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -186,7 +186,7 @@ internal Task AbortAsync()
             if (Features.Get<IConnectionInherentKeepAliveFeature>() == null)
             {
                 Debug.Assert(ProtocolReaderWriter != null, "Expected the ProtocolReaderWriter to be set before StartAsync is called");
-                _pingMessage = ProtocolReaderWriter.WriteMessage(PingMessage.Instance);
+                _pingMessage = ProtocolReaderWriter.WriteMessage(new PingMessage());
```

Comment:
So we decided that we're going to allocate an object per ping?

Reply:
This is initializing a field that lives for the entire connection.